service/internal/proctelemetry/config.go

```diff
@@ -92,3 +138,44 @@ func cardinalityFilter(kvs ...attribute.KeyValue) attribute.Filter {
 		return !filter.HasValue(kv.Key)
 	}
 }
+
+func initPrometheusExporter(prometheusConfig telemetry.Prometheus, asyncErrorChannel chan error) (sdkmetric.Reader, *http.Server, error) {
+	promRegistry := prometheus.NewRegistry()
+	if prometheusConfig.Host == nil {
+		return nil, nil, fmt.Errorf("host must be specified")
+	}
+	if prometheusConfig.Port == nil {
+		return nil, nil, fmt.Errorf("port must be specified")
+	}
+	wrappedRegisterer := prometheus.WrapRegistererWithPrefix("otelcol_", promRegistry)
+	// We can remove `otelprom.WithoutUnits()` when the otel-go start exposing prometheus metrics using the OpenMetrics format
+	// which includes metric units that prometheusreceiver uses to trim unit's suffixes from metric names.
+	// https://github.com/open-telemetry/opentelemetry-go/issues/3468
```

Comment:
Should we have a tracking issue for this on this repository (linking the one on opentelemetry-go)?

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector/issues/8043